src/mjolnir/pbfgraphparser.cc

```diff
@@ -288,6 +288,8 @@ struct graph_callback : public OSMPBF::Callback {
       sequence<OSMWayNode>::iterator element = (*way_nodes_)[current_way_node_index_];
       while (current_way_node_index_ < way_nodes_->size() &&
              (way_node = element = (*way_nodes_)[current_way_node_index_]).node.osmid_ == osmid) {
+        // we need to keep the duplicate flag that way parsing set
+        n.duplicate_ = way_node.node.duplicate_;
```

Comment:
because the way parser was only setting the nodeid for the nodes it referenced you can see below we just overwrite the whole thing with what we parsed from the nodes tags. however now that the way callback also writes this single bit saying its a duplicate we need to copy that forward before we overwrite the node data